### PR TITLE
add reducers and actions

### PR DESCRIPTION
**Pull request by Khanetmoi**
### This project milestone fulfills the following project requirements:

Write your book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

    Define action types for adding and removing a book.
    Set the initial state to be an empty array of books.
    Export Action Creators for your actions.
    Write your reducer and export it as default.
        Define state changes for the actions that you created.
        In case of unknown action - return the current state.

Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

    Define an action type for checking the status.
    Set the initial state to be an empty array of categories.
    Export Action Creators for your actions.
    Write your reducer and export it as default.
        For the check status action return a string "Under construction".
        In case of unknown action - return to the current state.

Configure the Redux Store (/src/redux/configureStore.js):

    Importing the necessary methods from Redux Toolkit.
    Combine both reducers into a root reducer by using configureStore function.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const REMOVE = 'bookstore/books/removeBook';
+
+const initialState = {
+  books: [],
+};
+
+export default function booksReducer(state = initialState, action) {
+  switch (action.type) {
+    case ADD:
+      return { ...state };
```
Comment (Felix45) [inline src/redux/books/books.js]: - [ Required ] I think that this statement would not ensure that a new book is added, remember that when `ADD` book action is dispatched a new book should be added to the redux state. To ensure that a new book is added, please consider refactoring this code as shown below
```JavaScript
  return [ ...state, action.payload ]
```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const ADD = 'bookstore/books/addBook';
+const REMOVE = 'bookstore/books/removeBook';
+
+const initialState = {
+  books: [],
+};
```
Comment (Felix45) [inline src/redux/books/books.js]: - [ Required ] Please consider setting the `initialState` to an array as specified in the project requirements, currently, the `initialState` is an object. You can do this by refactoring this code as shown below.
```JavaScript
 const initialState = []
```
## Project requirements
Screenshot from 2023-02-03 00-09-08

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+export default function booksReducer(state = initialState, action) {
+  switch (action.type) {
+    case ADD:
+      return { ...state };
+    case REMOVE:
+      return {
+        ...state,
+        items: state.items.filter((item) => item.id !== action.id),
+      };
```
Comment (Felix45) [inline src/redux/books/books.js]: - [ Required ] Once the `initialState` is set to an empty array then you'll need to refactor this line of code as shown below
```JavaScript
  return state.items.filter((item) => item.id !== action.id),
```

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const STATUS = 'bookstore/categories/statusChecked';
+
+const initialState = {
+  categories: [],
+};
```
Comment (Felix45) [inline src/redux/categories/categories.js]: - [ Required ] Please consider setting the `initialState` to an array as specified in the project requirements, currently, the `initialState` is an object. You can do this by refactoring this code as shown below.
```JavaScript
 const initialState = []
```
## Project requirements
Screenshot from 2023-02-03 00-23-06

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @Khanetmoi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [ ] No linter errors :white_check_mark: 
- [ ] Gitflow is used correctly :white_check_mark: 
- [ ] Descriptive pull request title and summary :white_check_mark: 


## Required Changes ♻️
- [ Required ] Please consider removing the texts highlighted in `red` from your readme file, these texts are supposed to be `instructions` on how to fill in your readme file. Once the sections are filled in correctly you need to remove them from the readme file.

Screenshot from 2023-02-03 00-30-24

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (franclobo) [inline src/redux/books/books.js]: - [ ] As required by the previous code reviewer, please set the initialState to an array 👇🏽 

> Please consider setting the initialState to an array as specified in the project requirements, currently, the initialState is an object. You can do this by refactoring this code as shown below.

File: src/redux/books/books.js (hunk 1 above)
Comment (franclobo) [inline src/redux/books/books.js]: - [ ] Please 🙏🏽 follow the suggestions from @Felix45 and complete all the changes required by the previous code reviewer before submitting the project again, you have limited chances to approve it. 👇🏽 

> [ Required ] I think that this statement would not ensure that a new book is added, remember that when ADD book action is dispatched a new book should be added to the redux state. To ensure that a new book is added, please consider refactoring this code as shown below

File: src/redux/books/books.js (hunk 3 above)
Comment (franclobo) [inline src/redux/books/books.js]: - [ ] Please 🙏🏽 follow the suggestions from @Felix45 and complete all the changes required by the previous code reviewer before submitting the project again, you have limited chances to approve it. 👇🏽 

> [ Required ] Once the initialState is set to an empty array then you'll need to refactor this line of code as shown below

File: src/redux/categories/categories.js (hunk 4 above)
Comment (franclobo) [inline src/redux/categories/categories.js]: - [ ] Please 🙏🏽 follow the suggestions from @Felix45 and complete all the changes required by the previous code reviewer before submitting the project again, you have limited chances to approve it. 👇🏽 

> 

> [ Required ] Please consider setting the initialState to an array as specified in the project requirements, currently, the initialState is an object. You can do this by refactoring this code as shown below.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Khanetmoi, 

# ♻️ Required Changes ♻️

soClose

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by franclobo — APPROVED** [review]
Hi @Khanetmoi, 

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**